app.py

```diff
@@ -132,20 +148,34 @@ def train_model(initial_run=False):
             metric_to_predict.label_config,
         )
 
+    data_queue.put(PREDICTOR_MODEL_LIST)
+
 
 if __name__ == "__main__":
+    # Queue to share data between the tornado server and the model training
+    predicted_model_queue = Queue()
+
     # Initial run to generate metrics, before they are exposed
-    train_model(initial_run=True)
+    train_model(initial_run=True, data_queue=predicted_model_queue)
 
-    # Start up the server to expose the metrics.
-    app = make_app()
+    # Set up the tornado web app
+    app = make_app(predicted_model_queue)
     app.listen(8080)
```

Comment:
Correct me if I'm wrong (I'm a Tornado noob), but aren't you starting the http server here? Reading [here](https://www.tornadoweb.org/en/stable/web.html#tornado.web.Application.listen) it seems like you want to "create an HTTPServer and call its TCPServer.bind/TCPServer.start methods directly" for multi processing.

Reply:
I think it starts the server when you do 'app.start()'

Reply:
Hmmm I always thought I was using the bind method but you're right